api/server-initialization.js

```diff
@@ -1,3 +1,4 @@
+const mongoose = require('mongoose');
```

Comment:
We don't know exactly which storage provider a user plugged-in. So doing it only for the mongodb case seems prone to errors. How about adding a finalizer method to the [StorageProvider](https://github.com/stellar-expert/operations-notifier/blob/93aa567c72c4c8a17cc2e14308ee80f9ef467326/persistence-layer/storage-provider.js) class and implementing it in derived providers?

Something like
```
finalize() {
    return mongoose.disconnect()
}
```

Then we can implement the graceful exit code in the provider-agnostic fashion, and it won't break when some implements, say, PostgreSQL plugin.